infra/modules/database/main.tf

```diff
@@ -0,0 +1,192 @@
+  key_id = "alias/aws/backup"
+}
+# create backup vault
+resource "aws_backup_vault" "postgresql" {
+  name        = "${var.database_name}-vault"
+  kms_key_arn = data.aws_kms_key.postgresql.arn
+}
+
+# create IAM role
+resource "aws_iam_role" "postgresql_backup" {
+  name               = "${var.database_name}-postgresql-backup"
+  assume_role_policy = data.aws_iam_policy_document.postgresql_backup.json
+}
+
+resource "aws_iam_role_policy_attachment" "postgresql_backup" {
+  role       = aws_iam_role.postgresql_backup.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSBackupServiceRolePolicyForBackup"
+}
+
+data "aws_iam_policy_document" "postgresql_backup" {
+  statement {
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["backup.amazonaws.com"]
+    }
+  }
+}
+# backup selection
+resource "aws_backup_selection" "postgresql_backup" {
+  iam_role_arn = aws_iam_role.postgresql_backup.arn
+  name         = "${var.database_name}-backup"
+  plan_id      = aws_backup_plan.postgresql.id
+
+  resources = [
+    aws_rds_cluster.postgresql.arn
+  ]
+}
+
+################################################################################
+# IAM role for enhanced monitoring
+################################################################################
+
+resource "aws_iam_role" "rds_enhanced_monitoring" {
+  name               = "${var.database_name}-rds-enhanced-monitoring"
+  assume_role_policy = data.aws_iam_policy_document.rds_enhanced_monitoring.json
+}
+
+resource "aws_iam_role_policy_attachment" "rds_enhanced_monitoring" {
+  role       = aws_iam_role.rds_enhanced_monitoring.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonRDSEnhancedMonitoringRole"
+}
+
+data "aws_iam_policy_document" "rds_enhanced_monitoring" {
+  statement {
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["monitoring.rds.amazonaws.com"]
+    }
+  }
+}
+
+################################################################################
+# Parameters for Query Logging
+################################################################################
+
+resource "aws_rds_cluster_parameter_group" "rds_query_logging" {
+  name        = var.database_name
+  family      = "aurora-postgresql13"
+  description = "Default cluster parameter group"
+
+  parameter {
+    name  = "log_statement"
+    value = "all" # ddl for template; none for wic
+  }
+
+  parameter {
+    name  = "log_min_duration_statement"
+    value = "1"
+  }
+}
+
+################################################################################
+# IAM role for user access
+################################################################################
+resource "aws_iam_policy" "db_access" {
+  name        = "${var.database_name}-db-access"
+  description = "Allows access to the database instance"
```

Comment:
This feels like an inaccurate description

IAM isn't required to access the database.
This is a permission to _modify_ and/or administer the database instance...

So, maybe
```terraform
   description = "Allows administration of the database instance"
```